### PR TITLE
gen crd's api docs

### PR DESCRIPTION
**Pull request by runzexia**
Signed-off-by: runzexia <runzexia@yunify.com>

### CODE REVIEW

File: tools/cmd/crd-doc-gen/main.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+	// tmp install s2i api group because
+	// cannot use Scheme (type *"kubesphere.io/kubesphere/vendor/k8s.io/apimachinery/pkg/runtime".Scheme) as type *"github.com/kubesphere/s2ioperator/vendor/k8s.io/apimachinery/pkg/runtime".Scheme in argument to "github.com/kubesphere/s2ioperator/pkg/apis/devops/install".Install
+	urlruntime.Must(s2iv1alpha1.AddToScheme(Scheme))
+	urlruntime.Must(Scheme.SetVersionPriority(s2iv1alpha1.SchemeGroupVersion))
+
+	tenantinstall.Install(Scheme)
+
+	mapper := meta.NewDefaultRESTMapper(nil)
+
+	mapper.AddSpecific(tenantv1alpha1.SchemeGroupVersion.WithKind(tenantv1alpha1.ResourceKindWorkspace),
```
Comment (zryfish) [inline tools/cmd/crd-doc-gen/main.go]: what about servicemesh related crds?

File: tools/cmd/crd-doc-gen/main.go [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+			s2iv1alpha1.SchemeGroupVersion.WithResource(s2iv1alpha1.ResourcePluralS2iBuilderTemplate),
+			s2iv1alpha1.SchemeGroupVersion.WithResource(s2iv1alpha1.ResourcePluralS2iBuilder),
+		},
+		Mapper: mapper,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	filename := build.Default.GOPATH + "/src/kubesphere.io/kubesphere/api/openapi-spec/swagger.json"
```
Comment (zryfish) [inline tools/cmd/crd-doc-gen/main.go]: why use absolute path here? Go module doesn't require project folder placed under GOPATH.

File: tools/cmd/crd-doc-gen/main.go (hunk 1 above)
Comment (runzexia) [reply tools/cmd/crd-doc-gen/main.go]: Currently lacking the isio-related go language openapi structure, I need to submit some code to github/knative/pkg to generate the code.

Comment (runzexia) [reply tools/cmd/crd-doc-gen/main.go]: https://github.com/kubesphere/kubesphere/pull/609/files#diff-b4163ba54c93a4371a0a28df03f28b28R94

**Comment by runzexia** [comment]
fix merge confilcts & add network crd doc gen

**Pull request merged**